contracts/zero-ex/tests/Multiplex.t.sol

```diff
@@ -0,0 +1,1453 @@
+import "../contracts/test/TestMintTokenERC20Transformer.sol";
+import "../contracts/test/tokens/TestMintableERC20Token.sol";
+import "../contracts/test/integration/TestUniswapV2Factory.sol";
+import "../contracts/test/integration/TestUniswapV2Pool.sol";
+import "../contracts/test/integration/TestUniswapV3Factory.sol";
+import "../contracts/test/integration/TestUniswapV3Pool.sol";
+import "../contracts/test/integration/TestLiquidityProvider.sol";
+
+import "@0x/contracts-erc20/contracts/src/v06/WETH9V06.sol";
+
+contract Multiplex is Test, ForkUtils, TestUtils {
+    uint256 private constant MAX_UINT256 = type(uint256).max;
+    uint256 private constant HIGH_BIT = 2 ** 255;
+    uint24 private constant POOL_FEE = 1234;
+
+    bytes32 private constant RFQ_ORDER_FILLED_SIG =
+        keccak256(
+            "RfqOrderFilled("
+            "bytes32,"
+            "address,"
+            "address,"
+            "address,"
+            "address,"
+            "uint128,"
+            "uint128,"
+            "bytes32"
+            ")"
+        );
+
+    bytes32 private constant OTC_ORDER_FILLED_SIG =
+        keccak256(
+            "OtcOrderFilled("
+            "bytes32,"
+            "address,"
+            "address,"
+            "address,"
+            "address,"
+            "uint128,"
+            "uint128"
+            ")"
+        );
+
+    bytes32 private constant EXPIRED_RFQ_ORDER_SIG =
+        keccak256(
+            "ExpiredRfqOrder("
+            "bytes32,"
+            "address,"
+            "uint64"
+            ")"
+        );
+
+    bytes32 private constant EXPIRED_OTC_ORDER_SIG =
+        keccak256(
+            "ExpiredOtcOrder("
+            "bytes32,"
+            "address,"
+            "uint64"
+            ")"
+        );
+
+    bytes32 private constant TRANSFER_SIG =
+        keccak256(
+            "Transfer("
+            "address,"
+            "address,"
+            "address,"
+            "uint256"
+            ")"
+        );
+
+    bytes32 private constant MINT_TRANSFORM_SIG =
+        keccak256(
+            "MintTransform("
+            "address,"
+            "address,"
+            "address,"
+            "address,"
+            "bytes,"
+            "uint256,"
+            "uint256"
+            ")"
+        );
+
+    struct LogDescription {
+        bytes32[] topics;
+        bytes data;
+        bytes dataMask;
+    }
+
+    DeployZeroEx.ZeroExDeployed private zeroExDeployed;
+    IFlashWallet private flashWallet;
+    IERC20TokenV06 private shib;
+    IERC20TokenV06 private dai;
+    IERC20TokenV06 private zrx;
+    IEtherTokenV06 private weth;
+
+    TestUniswapV2Factory private sushiFactory;
+    TestUniswapV2Factory private uniV2Factory;
+    TestUniswapV3Factory private uniV3Factory;
+    TestLiquidityProvider private liquidityProvider;
+    uint256 private transformerNonce;
+
+    address private signerAddress;
+    uint256 private signerKey;
+
+    modifier explain(string memory testName) {
```

Comment:
Do we consume this log output in a GitHub action or something? Seems needlessly noisy to me. We definitely want the relevant output when the test fails, though

Reply:
This is here because I was trying to find a good way to label the tests with the descriptions from the original typescript tests. This is just about as noisy as running the typescript tests we use right now, but I do agree that maybe even the typescript tests are a bit noisy. Unsure what a good alternative would be though.

It doesn't seem right to me to shoehorn an English sentence into the already-long name of a function, like `test_multiplexBatchSellTokenForToken_revertsIfMinBuyAmountIsNotSatisfied` especially if we want to be any more descriptive or include special characters. It encourages writers of tests to make the descriptions more abbreviated, truncated, and unclear. So, logging the description of the test seems to be a good solution. I agree that only logging on failure would be better, but afaik we can't reliably do that with forge (we can if we always use try/catch in tests, but if we rely on reverts to fail a test then we can't).